### DISCUSSION TITLE
One UI with all capture fields, rather than one UI per field

### DISCUSSION THREAD

**Discussion by robosquiggles** [chhoumann/quickadd#268] [Ideas]
I love this plugin! I use it all the time. Thanks so much for your contribution to the community!

One thing I tend to struggle with is entering all the information in the order it asks me for it. For example, I have an "Add task to daily note" Capture, which takes several fields as input and adds a task to my daily note file for that day. Here's the capture format:

```markdown
- [ ] #task {{VALUE:task description}} ⏰ {{VDATE:due date,YYYY-MM-DD}} {{VDATE:reminder time, HH:mm}} 📅 {{VDATE:due date,YYYY-MM-DD}}
```

Usually, when I use this Capture I'm thinking "I have a new take, I better write it down before I lose the thought!" When I trigger the QuickAdd, however, I get one UI with a single field that asks for `due date`, then one for `reminder time`, then finally the `description`. This _works_, but there are two problems:
1. The order in which I am asked for input is reversed from what I entered in the Capture Formant, which is a bit confusing.
2. It would be nice to be able to pick the order of the fields that pop up, or have them pop up in the "correct" order (ie. put the description first)
3. I'd also love to be able to go back to edit a field if I messed it up.

I'm picturing a basic GUI with one field for every field I wish to enter. This would make it so that I could tab through each field and enter information in whatever order I choose. I think this would offer a much smoother user experience!

Thoughts?

Thanks again for the fabulous plugin!

**Comment by taglia** [comment]
I also love the plugin, and I have the same issue that @robosquiggles is describing. I always end up typing the _task name_ in the _due date_ field, pressing enter, and having to restart from scratch. Likewise, I love the suggestion above, but even just being able to choose the order in which the fields appear (like: I want _task name_ first, then _due date_) would be great!

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

The GUI you sketched exists now: it is called one-page inputs, and it shipped in 2.10.0. Toggle **One-page input for choices** in Settings -> QuickAdd (it is behind a toggle marked Beta, but it is stable in daily use), or use the per-choice **One-page input override** if you only want it for some choices.

With it on, your task capture collects everything in a single form: one text field for `task description` and date inputs for the `{{VDATE}}`s, in the order they appear in your format - so the description comes first, you can tab between fields, and you can fix a field before submitting instead of restarting. That covers all three of your points (order, choosing where to start, and going back to edit).

Docs: https://quickadd.obsidian.guide/docs/Advanced/onePageInputs

Closing as resolved.
